### PR TITLE
in-app messaging docs Gen 2

### PR DESCRIPTION
**Pull request by ykethan**
#### Description of changes:
add in-app messaging docs to Gen 2

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  };
+}
+
+A campaign is a messaging initiative that engages a specific audience [segment](https://docs.aws.amazon.com/pinpoint/latest/userguide/segments.html). A campaign sends tailored messages according to a schedule that you define. You can use the AWS console to create a campaign that sends messages through any single channel that is supported by Amazon Pinpoint: Mobile Push, In-App, Email, SMS or Custom channels. [Learn how to create a campaign using Amazon Pinpoint](https://docs.aws.amazon.com/pinpoint/latest/userguide/campaigns.html) to continue integrating in-app messages in your app with Amplify.
+
+1. Login to the [AWS Console](https://console.aws.amazon.com/console/home), and Search for **Pinpoint**.
+2. Click on your project from the list of available project. Your project name would have been setup with the CLI as your Pinpoint resource name when the `amplify add notifications` command was used.
+3. Click on **Campaigns** from the left navigation menu, and then click on **Create a campaign**
+
+   ![In-App-Messaging](/images/lib/in-app-messaging/create-pinpoint-campaign.png)
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx:35]: Could we please get a more descriptive alt text for the image?

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+   2. Channel: **In-App messaging**
+   3. set prioritization: **Fairly important**
+
+5. Click on the **Create a segment** radio button, and add a name for your segment, and then click **Next**.
+
+   1. You can add as many segments as needed to the campaign. For this quickstart, you can use **Include any audiences** under the _Segment group 1_ section.
+   2. You can add a criteria to your segments to ensure that audiences that satisfy that criteria can receive the in-app message.
+   3. If you see an error message titled _Segment might include multiple channels_, click **I understand** to proceed.
+
+   ![In-App-Messaging](/images/lib/in-app-messaging/choose-campaign-segment.png)
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx:49]: Could we please get a more descriptive alt text for the image?

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+<InlineFilter filters={['react-native']}>
+   <Callout informational>
+      As React Native does not support SVG rendering out of the box, Amplify cannot render SVG images by default. For SVG image support with In-App Messaging a custom UI implementation is required.
+   </Callout>
+</InlineFilter>
+
+For this tutorial you can create a simple message as shown below. Customers in your application will see the same message once the event is triggered.
+
+![In-App-Messaging](/images/lib/in-app-messaging/in-app-message-details.png)
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx:70]: Could we please get a more descriptive alt text for the image?

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+8. Once you have finished customizing your in-app message, click on **Next**.
+9. Under _Trigger events_, add the name of the analytics trigger that will be sent from your frontend app. 
+   - You have the ability to customize the trigger to allow only certain attributes or metrics that are passed with the analytics event to trigger the in-app message. (Optional)
+
+![In-App-Messaging](/images/lib/in-app-messaging/campaign-setup.png)
+
+10. By default, the number of messages shown per session is 1. You can update this threshold during campaign setup. 
+
+![In-App-Messaging](/images/lib/in-app-messaging/campaign-settings.png)
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx:80]: Could we please get a more descriptive alt text for the image?

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+For this tutorial you can create a simple message as shown below. Customers in your application will see the same message once the event is triggered.
+
+![In-App-Messaging](/images/lib/in-app-messaging/in-app-message-details.png)
+
+8. Once you have finished customizing your in-app message, click on **Next**.
+9. Under _Trigger events_, add the name of the analytics trigger that will be sent from your frontend app. 
+   - You have the ability to customize the trigger to allow only certain attributes or metrics that are passed with the analytics event to trigger the in-app message. (Optional)
+
+![In-App-Messaging](/images/lib/in-app-messaging/campaign-setup.png)
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx:76]: Could we please get a more descriptive alt text for the image?

**Review by timngyn — COMMENTED** [review]
Hey @ykethan, could we get more descriptive text for the images? 

Some resources to help write alt text:

https://webaim.org/techniques/alttext/
https://www.a11yproject.com/posts/are-you-making-these-five-mistakes-when-writing-alt-text/

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/clear-messages/index.mdx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+export function getStaticProps(context) {
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+Once messages have been synced to your user’s device, `clearMessages()` can be used to clear the synced messages.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/clear-messages/index.mdx:29]:
```suggestion
Once messages have been synced to your user's device, `clearMessages()` can be used to clear the synced messages.
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/clear-messages/index.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+await clearMessages();
+```
+
+<Callout informational>
+  NOTE: If your app has authentication implemented, we recommend calling
+  clearMessages() in between user log-ins to remove messages targeted for
+  specific user segments. This is especially important if you anticipate your
+  application will be used in shared device scenarios.
+</Callout>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/clear-messages/index.mdx:42]:
```suggestion
<Callout informational>

**Note:** If your app has authentication implemented, we recommend calling `clearMessages()` in between user log-ins to remove messages targeted for specific user segments. This is especially important if you anticipate your application will be used in shared device scenarios.
  
</Callout>
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+![In-App-Messaging](/images/lib/in-app-messaging/campaign-settings.png)
+
+11. Review your campaign, and then click on **Launch campaign**.
+
+Your campaign is now setup, and you are ready to start integrating the In-App Messaging functionality into your app.
+
+<Callout informational>
+
+Note: Campaign start time must be at least 15 minutes in future. In-app messages can only be synced to local device once the campaign becomes active (Status should be "In Progress" in the campaigns screen of the Pinpoint console).
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/create-campaign/index.mdx:88]:
```suggestion
**Note:** Campaign start time must be at least 15 minutes in future. In-app messages can only be synced to local device once the campaign becomes active (status should be "In Progress" in the campaigns screen of the Pinpoint console).
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/display-messages/index.mdx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+      meta
+    }
+  };
+}
+
+In-app messages are displayed when an In-App Messaging or analytics event is sent and matches the criteria set forth by your active In-App Messaging campaigns.
+
+## Analytics event
+
+Now that messages have been synced to your users’ devices, Amplify In-App Messaging will allow you to start displaying them with Amplify Analytics events with no additional integration steps. Any events you record or are already recording using the Analytics' `record` API are automatically picked up and processed by In-App Messaging. If the event matches the attributes and criteria defined in an in-app message, that message will be displayed.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/display-messages/index.mdx:33]:
```suggestion
Now that messages have been synced to your users' devices, Amplify In-App Messaging will allow you to start displaying them with Amplify Analytics events with no additional integration steps. Any events you record or are already recording using the Analytics' `record` API are automatically picked up and processed by In-App Messaging. If the event matches the attributes and criteria defined in an in-app message, that message will be displayed.
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/identify-user/index.mdx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+      longitude: 0.0
+    },
+    metrics: {
+      // E.g. logins: 157
+    },
+  },
+};
+
+await identifyUser(identifyUserInput);
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/identify-user/index.mdx:68]:
````suggestion
```js
import { identifyUser } from 'aws-amplify/in-app-messaging';

await identifyUser({
  userId: '', // E.g. user-id
  userProfile: {
    email: '', // E.g. example@service.com
    name: '', // E.g. name-of-the-user
    plan: '' // E.g. plan-they-subscribe-to
    customProperties: {
      // E.g. hobbies: ['cooking', 'knitting'],
    },
    demographic: {
      appVersion: '',
      locale: '', // E.g. en_US
      make: '', // E.g. Apple
      model: '', // E.g. iPhone
      modelVersion: '', // E.g. 13
      platform: '', // E.g. iOS
      platformVersion: '', // E.g. 15
      timezone: '' // E.g. Americas/Los_Angeles
    },
    location: {
      city: '', // E.g. Seattle
      country: '', // E.g. US,
      postalCode: '', // E.g. 98121
      region: '', // E.g. WA
      latitude: 0.0,
      longitude: 0.0
    },
    metrics: {
      // E.g. logins: 157
    },
  },
});
```
````

defining this object inside the function argument grants type hints. We could also keep the object and import the type used for the arg

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/identify-user/index.mdx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    optOut: ''  // Either ALL or NONE
+    userAttributes: {
+      // E.g. interests: ['soccer', 'shoes'],
+    }
+  },
+
+};
+
+await identifyUser(identifyUserInput);
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/identify-user/index.mdx:90]:
````suggestion
```js
import { identifyUser } from 'aws-amplify/in-app-messaging';

await identifyUser({
  userId: '', // E.g. user-id
  options: {
    address: '' // E.g. A device token or email address
    optOut: ''  // Either ALL or NONE
    userAttributes: {
      // E.g. interests: ['soccer', 'shoes'],
    }
  },
});
```
````

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/integrate-application/index.mdx [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+
+<Accordion title="Instructions for React Native version 0.72 and below">
+
+  `@aws-amplify/react-native` requires a minimum iOS deployment target of `13.0` if you are using `react-native` version less than or equal to `0.72`. Open the _Podfile_ located in the _ios_ directory and update the `target` value:
+
+  ```diff
+   - platform :ios, min_ios_version_supported
+   + platform :ios, 13.0
+   ```
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/integrate-application/index.mdx:41]:
```suggestion
`@aws-amplify/react-native` requires a minimum iOS deployment target of `13.0` if you are using `react-native` version less than or equal to `0.72`. Open the _Podfile_ located in the _ios_ directory and update the `target` value:

```diff
- platform :ios, min_ios_version_supported
+ platform :ios, 13.0
```

```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/integrate-application/index.mdx [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+npm install @aws-amplify/react-native @react-native-community/netinfo @react-native-async-storage/async-storage
+```
+
+### Install Amplify UI for React Native and its dependencies
+
+Although Amplify In-App Messaging can be used as a standalone JavaScript library, this guide will show you how to use it together with Amplify UI which currently supports integration with React and React Native to get started quickly.
+
+<Callout>
+
+  Learn more about Amplify In-App Messaging UI and how to fully unlock its capabilities here: [Amplify UI for In-App Messaging](https://ui.docs.amplify.aws/react-native/connected-components/in-app-messaging)
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/integrate-application/index.mdx:54]:
```suggestion
Learn more about Amplify In-App Messaging UI and how to fully unlock its capabilities here: [Amplify UI for In-App Messaging](https://ui.docs.amplify.aws/react-native/connected-components/in-app-messaging)
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+Amplify allows interacting with In-App Messaging APIs, which enables you to send messages to your app users. In-App Messaging is a powerful tool to engage with your users and provide them with relevant information. 
+A campaign is a messaging initiative that engages a specific audience segment. A campaign sends tailored messages according to a schedule that you define. You can use the [AWS Cloud Development Kit (AWS CDK)](https://docs.aws.amazon.com/cdk/latest/guide/home.html) to create a campaign that sends messages through any single channel that is supported by Amazon Pinpoint: Mobile Push, In-App, Email, SMS or Custom channels.
+
+The following is an example utilizing the AWS CDK to create the In-App Messaging resource powered by [Amazon Pinpoint](https://aws.amazon.com/pinpoint/).
+
+<Callout informational>
+
+**Note**: Campaign start time must be at least 15 minutes in future. In-app messages can only be synced to local device once the campaign becomes active (Status should be "In Progress" in the campaigns screen of the Pinpoint console).
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx:36]:
```suggestion
**Note:** Campaign start time must be at least 15 minutes in future. In-app messages can only be synced to local device once the campaign becomes active (Status should be "In Progress" in the campaigns screen of the Pinpoint console).
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+
+```ts title="amplify/backend.ts"
+import {
+  CfnApp,
+  CfnInAppTemplate,
+  CfnCampaign,
+  CfnSegment,
+} from "aws-cdk-lib/aws-pinpoint";
+import { Policy, PolicyStatement } from "aws-cdk-lib/aws-iam";
+import { Stack } from "aws-cdk-lib";
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx:48]:
```suggestion
import { defineBackend } from "@aws-amplify/backend";
import {
  CfnApp,
  CfnInAppTemplate,
  CfnCampaign,
  CfnSegment,
} from "aws-cdk-lib/aws-pinpoint";
import { Policy, PolicyStatement } from "aws-cdk-lib/aws-iam";
import { Stack } from "aws-cdk-lib/core";
import { auth } from "./auth/resource";
import { data } from "./data/resource";
```

added the resource imports and swapped the Stack import to "/core"

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx [hunk 16]
```diff
@@ -0,0 +1,182 @@
...
+});
+
+// create a segment 
+const mySegment = new CfnSegment(inAppMessagingStack, "Segment", {
+  applicationId: pinpoint.ref,
+  name: "mySegment",
+});
+
+// create a campaign with event and in-app message template
+new CfnCampaign(inAppMessagingStack, "MyCampaign", {
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx:70]:
```suggestion
new CfnCampaign(inAppMessagingStack, "Campaign", {
```

for consistency with the others

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+
+### Initialize In-App Messaging
+
+To finish setting up your application with Amplify, you need to configure it using the `configure` API. Next, to interact with In-App Messaging APIs, you need to first initialize In-App Messaging by calling the `initializeInAppMessaging` API directly imported from the `in-app-messaging` sub-path. This is required to be called as early as possible in the app lifecycle.
+
+<InlineFilter filters={["javascript", "angular", "react", "vue", "react-native"]}>
+```js
+import { Amplify } from 'aws-amplify';
+import amplifyconfig from '../amplifyconfiguration.json';
+import { initializeInAppMessaging } from 'aws-amplify/in-app-messaging';
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx:154]:
```suggestion
import { initializeInAppMessaging } from 'aws-amplify/in-app-messaging';
import amplifyconfig from '../amplifyconfiguration.json';
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+Amplify.configure(amplifyconfig);
+initializeInAppMessaging();
+```
+</InlineFilter>
+
+<InlineFilter filters={['nextjs']}> 
+```js
+import { Amplify } from 'aws-amplify';
+import amplifyconfig from '@/amplifyconfiguration.json';
+import { initializeInAppMessaging } from 'aws-amplify/in-app-messaging';
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx:165]:
```suggestion
import { initializeInAppMessaging } from 'aws-amplify/in-app-messaging';
import amplifyconfig from '@/amplifyconfiguration.json';
```

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/sync-messages/index.mdx [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+```js
+import { syncMessages } from 'aws-amplify/in-app-messaging';
+
+await syncMessages();
+```
+
+<Callout informational>
+  **Note**: Syncing messages will always overwrite existing messages currently on
+  the user's device so they are always up to date when the sync is performed.
+</Callout>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/sync-messages/index.mdx:40]:
```suggestion
<Callout informational>

**Note**: Syncing messages will always overwrite existing messages currently on the user's device so they are always up to date when the sync is performed.

</Callout>
```

**Review by josefaidt — DISMISSED** [review]

**Review by timngyn — DISMISSED** [review]

**Review by aspittel — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx [hunk 20]
```diff
@@ -0,0 +1,190 @@
...
+// apply the policy to the authenticated and unauthenticated roles
+backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(pinpointPolicy);
+backend.auth.resources.unauthenticatedUserIamRole.attachInlinePolicy(pinpointPolicy);
+
+// patch the custom Pinpoint resource to the expected output configuration
+backend.addOutput({
+  notifications: {
+    amazon_pinpoint_app_id: pinpoint.ref,
+    aws_region: Stack.of(pinpoint).region,
+    channels: ["IN_APP_MESSAGING"],
```
[the remark this answers is not in the document]
Comment (ykethan) [reply src/pages/[platform]/build-a-backend/add-aws-services/in-app-messaging/set-up-in-app-messaging/index.mdx:141]: the type needs to be improved on this, the `channels` were missing added that which should not cause any issues.

**Pull request merged**